### PR TITLE
[DX] Report files with start with <?= short php tag

### PR DESCRIPTION
**Pull request by samsonasik**
@TomasVotruba @staabm here on progress to report files with start with `<?=` short php tag

Ref https://github.com/rectorphp/rector-src/pull/6068#issuecomment-2197162261
Ref https://3v4l.org/tHtB2

**For direct files**

Screenshot 2024-06-29 at 10 21 57

**For inside directory**

Screenshot 2024-06-29 at 10 52 49

### CODE REVIEW

**Comment by samsonasik** [comment]
implemented 🎉

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: src/Reporting/MissConfigurationReporter.php [hunk 1]
```diff
@@ -54,4 +54,23 @@ public function reportVendorInPaths(array $filePaths): void
...
     }
+
+    public function reportStartWithShortOpenTag(): void
+    {
+        $files = SimpleParameterProvider::provideArrayParameter(Option::SKIPPED_START_WITH_SHORT_OPEN_TAG_FILES);
+        if ($files === []) {
+            return;
+        }
+
+        $suffix = count($files) > 1 ? 's' : '';
```
Comment (paulbalandan) [inline src/Reporting/MissConfigurationReporter.php:65]: Grammar fix: (I can't edit properly on mobile so sending here)
```php
$suffix = count($files) > 1 ? 's were' : ' was'; 
$fileList = implode(PHP_EOL, $files);
$this->symfonyStyle->warning(sprintf( 
    'The following file%s skipped as starting with short open tag. Migrate to long open PHP tag first: %s%s',
    $suffix,
    PHP_EOL . PHP_EOL,
    $fileList 
));
```

**Comment by staabm** [comment]
Does the problem also happen when a short open tag is used somewhere within the file?

File: src/Reporting/MissConfigurationReporter.php (hunk 1 above)
Comment (samsonasik) [reply src/Reporting/MissConfigurationReporter.php:65]: I also on mobile, will check when I am back

**Comment by samsonasik** [comment]
It only verify content file start with for safest, as in the middle can be a string

File: src/FileSystem/FilesFinder.php [hunk 2]
```diff
@@ -48,7 +51,14 @@ public function findInDirectoriesAndFiles(array $source, array $suffixes = [], b
 
         $filteredFilePaths = array_filter(
             $filteredFilePaths,
-            fn (string $file): bool => ! $this->isStartWithShortPHPTag(FileSystem::read($file))
+            function (string $file): bool {
+                if ($this->isStartWithShortPHPTag(FileSystem::read($file))) {
```
Comment (staabm) [inline src/FileSystem/FilesFinder.php:55]: We should have a test whether this works for files using a byte order mark (BOM)

Comment (TomasVotruba) [reply src/FileSystem/FilesFinder.php:55]: That could be done in separate go. Care for PR? :)

**Comment by TomasVotruba** [comment]
Thanks :+1:

File: src/Reporting/MissConfigurationReporter.php (hunk 1 above)
Comment (samsonasik) [reply src/Reporting/MissConfigurationReporter.php:65]: https://github.com/rectorphp/rector-src/pull/6079

**Pull request merged**